go-app-ussd_popi_rapidpro.js

```diff
@@ -478,14 +478,28 @@ go.app = function() {
             baby_dob2 = self.dateformat(baby_dob2);
             baby_dob3 = self.dateformat(baby_dob3);
             edd = self.dateformat(edd);
-            var context = {
-                dobs: _.map(_.filter([
-                    new moment(edd),
-                    new moment(baby_dob1),
-                    new moment(baby_dob2),
-                    new moment(baby_dob3),
-                ], _.method("isValid")), _.method("format", "DD-MM-YYYY")).join(", ") || $("None")
-             };
+            var context;
+
+            if (edd){
+                context = {
+                    dobs: _.map(_.filter([
+                        new moment(edd),
+                        new moment(baby_dob1),
+                        new moment(baby_dob2),
+                        new moment(baby_dob3),
+                    ], _.method("isValid")), _.method("format", "DD-MM-YYYY")).join(", ") || $("None")
+                 };
+             }
+             else {
+                context = {
+                    dobs: _.map(_.filter([
+                        new moment(baby_dob1),
+                        new moment(baby_dob2),
+                        new moment(baby_dob3),
+                    ], _.method("isValid")), _.method("format", "DD-MM-YYYY")).join(", ") || $("None")
+                 };
+             }
+
```

Comment:
@Buhle79 I now remember why we didn't consider a (dob but no edd) option because a contact with dob would have an edd and at the time we weren't clearing the edd field in RP (not sure if we now do). But it's good to have this check.